### DISCUSSION TITLE
Using the same wallet for multiple coins

### DISCUSSION THREAD

**Discussion by rdong8** [hdwallet-io/python-hdwallet#106] [Q&A] [answered]
Noob question but when you export from ie. your Phantom wallet, it gives you a single seed phrase even though Phantom supports multiple networks. However here, you have to pass `cryptocurrency=...` in the HDWallet constructor, why is that?

Also, if you set say `cryptocurrency=Bitcoin` when creating the wallet, it then allows you to do `from_derivation(BIP44Derivation(coin_type=Ethereum))` and that seems to work as well. However, the resulting address is an Ethereum address and not a Bitcoin address. 

If I wanted to do it like Phantom and just have a single seed phrase or keypair for multiple cryptocurrencies, how would I do that in this library?

**Comment by meherett** [comment, marked as answer]
If you're looking for a solution similar to how Phantom handles a single seed phrase across multiple networks, you can refer to the Phantom wallet client implementation here: https://github.com/talonlab/python-hdwallet/blob/master/clients/phantom.py. 
This should provide helpful insights. 

Thanks!

**Reply by rdong8** [reply]
Much appreciated! How should one generate a wallet without first associating it with a particular coin type? The application in question provides users with a custodial wallet intended for multichain use. I'd assume I should store only the xprivate key or mnemonic in the backend, and when an address on a particular chain is requested, I create an `HDWallet` object using the coin type of that chain and pass in the xprivate key/mnemonic (as in the example) to obtain an address on that chain. But it seems like to generate an xprivate key or mnemonic you need to set a coin type upfront to be able to create the wallet object.

**Comment by rdong8** [comment]
Also, how can I extract the mnemonic from a wallet created using the xprivate key?

**Reply by meherett** [reply]
You can't extract a mnemonic from an XPrivateKey because the mnemonic is converted into a root seed through a one-way process. The XPrivateKey is derived from this root seed, but the process is irreversible by design for security. Always back up your mnemonic, as it cannot be recovered from the XPrivateKey.
